components/ai_chat/core/common/mojom/page_content_extractor.mojom

```diff
@@ -13,10 +13,17 @@ enum PageContentType {
   VideoTranscriptVTT
 };
 
+// Configuration for YouTube InnerTube API
+struct YoutubeInnerTubeConfig {
+  string api_key;
+  string video_id;
+};
+
 // Items either have content or a url to retrieve content
 union PageContentData {
   string content;
   url.mojom.Url content_url;
+  YoutubeInnerTubeConfig youtube_inner_tube_config;
```

Comment:
I thought I made a comment about making this a union with `content_url` because these are apparently mutually exclusive? https://github.com/brave/brave-core/pull/29619/files